h/api/test/auth_test.py

```diff
@@ -28,3 +31,150 @@ def test_translate_annotation_principals(p_in, p_out):
     result = auth.translate_annotation_principals(p_in)
 
     assert set(result) == set(p_out)
+
+
+generate_bearer_token_fixtures = pytest.mark.usefixtures('jwt')
+
+
+@generate_bearer_token_fixtures
+def test_generate_bearer_token_calls_encode(jwt):
+    """It should pass the right arguments to encode()."""
+    before = datetime.datetime.utcnow()
+    request = mock_request()
+
+    auth.generate_bearer_token(request, 3600)
+
+    assert jwt.encode.call_args[0][0]['sub'] == 'acct:testuser@hypothes.is', (
+        "It should encode the userid as 'sub'")
+    after = datetime.datetime.utcnow() + datetime.timedelta(seconds=3600)
```

Comment:
This is fine, and I'll merge it, but here's another way of writing this test that personally I think would be much clearer and easier to maintain:

```python
class ApproximateMatcher(object):
    def __init__(self, value, delta):
        self.value = value
        self.delta = delta
    def __eq__(self, other):
        return self.value - self.delta < other < self.value + self.delta

def approximate_datetime(when):
    return ApproximateMatcher(when, delta=datetime.timedelta(seconds=2))
...

jwt.encode.assert_called_once_with({
    'sub': 'acct:testuser@hypothes.is',
    'iss': 'expected-client-id',
    'aud': 'expected-audience',
    'iat': approximate_datetime(now),
    'exp': approximate_datetime(now + datetime.timedelta(seconds=3600)),
}, secret, algorithm='HS256')
```